src/interpreter/contract.rs

```diff
@@ -23,15 +24,27 @@ where
     pub(crate) fn contract_balance(&mut self, ra: RegisterId, b: Word, c: Word) -> Result<(), RuntimeError> {
         Self::is_register_writable(ra)?;
 
-        if b > VM_MAX_RAM - AssetId::LEN as Word || c > VM_MAX_RAM - ContractId::LEN as Word {
+        let bx = b
+            .checked_add(AssetId::LEN as Word)
+            .ok_or_else(|| PanicReason::ArithmeticOverflow)?;
+
+        let cx = c
+            .checked_add(ContractId::LEN as Word)
+            .ok_or_else(|| PanicReason::ArithmeticOverflow)?;
+
+        //if bx or cx is above usize::MAX then it cannot be safely cast to usize,
+        // check the tighter bound
+        let min_comparison_value = cmp::min(VM_MAX_RAM, usize::MAX as Word);
```

Comment:
consider declaring this a `const` because it will be compiler to a constant anyway but then we see that happening in the code.

Also, this check might be overkill because `usize` is `u32` minimum and our VM memory won't ever be close to that value. However, doesn't hurt to check!